### PR TITLE
Don't autosave if file has been moved or deleted

### LINKED ISSUE #9: Autosaves moved files
Moving a file in the tree view causes an autosave.

Refs atom/tree-view#123

### PR DESCRIPTION
**Pull request by balexand**
Fixes #9

### CODE REVIEW

**Comment by balexand** [comment]
Please let me know if there's anything that I can do to help get this merged. This seems like an obvious fix since autosave should only update existing files, not create new ones.

I created a quick test that effectively reproduces the problem. I don't love that it copies a file back and forth. Maybe someone more experienced in the atom test environment can come up with a more elegant test.

Thanks!

**Comment by balexand** [comment]
It looks like [`fs.existsSync` and `fs.exists`](https://nodejs.org/api/fs.html#fs_fs_existssync_path) are deprecated. I've switched to `fs.statSync` and added the necessary exception handling.

**Comment by balexand** [comment]
I was taking advantage of the global `fs` variable, which was accidentally available prior to 0.199.0.

2015-05-15 at 2 45 pm

I've pushed a commit fixing this. While I was doing this, I realized that Atom has an [`fs-plus`](https://github.com/atom/fs-plus) module that contains `fs.isFileSync`. I'm now using this method.

I'll squash commits now. I hope this gets merged soon. Thanks!

**Comment by philipgiuliani** [comment]
@kevinsawicki Whats with this PR? :)

**Comment by maxbrunsfeld** [comment]
Looks good to me.

**Comment by freemanoid** [comment]
Any blockers? Please, tell us if there're any problems with this PR so we can fix and merge it.

/cc @mnquintana

File: lib/autosave.coffee [hunk 1]
```diff
@@ -34,6 +35,7 @@ module.exports =
     return unless atom.config.get('autosave.enabled')
     return unless paneItem?.getURI?()?
     return unless paneItem?.isModified?()
+    return unless paneItem?.getPath()? and fs.isFileSync(paneItem.getPath())
```
Comment (mnquintana) [inline lib/autosave.coffee]: What happens here if a pane item doesn't implement `::getPath()`? Won't it throw a `TypeError: undefined is not a function`? I'd add another existential operator guard here like the above lines just to be safe:

```coffee
paneItem?.getPath?()?
```

File: spec/autosave-spec.coffee [hunk 2]
```diff
@@ -46,6 +47,22 @@ describe "Autosave", ->
         document.body.focus()
         expect(initialActiveItem.save).toHaveBeenCalled()
 
+      it "suppresses autosave if the files doesn't exist", ->
```
Comment (mnquintana) [inline spec/autosave-spec.coffee]: :memo: `doesn't` -> `do not`

**Comment by balexand** [comment]
@mnquintana I've addressed your suggestions. In the spec wording, I changed "files" to be singular.

Please let me know if there's anything else I can do to help get this merged. Thanks!

**Comment by mnquintana** [comment]
Awesome, thanks so much for adding this :boat:

**Comment by balexand** [comment]
@mnquintana Thanks! :sparkles:

**Pull request merged**